### PR TITLE
Replace int1e in PCM and QMMM

### PR DESCRIPTION
**Pull request by henryw7**
An overlap improvement of 10% in SCF with solvent and 20% in Hessian calculation with solvent (31 atoms, B3LYP/def2-SVP, C-PCM)

### CODE REVIEW

File: gpu4pyscf/solvent/pcm.py [hunk 1]
```diff
@@ -393,19 +394,23 @@ def _get_v(self, dms):
...
         '''
         nset = dms.shape[0]
-        ngrids = self.surface['grid_coords'].shape[0]
+        charge_exp  = self.surface['charge_exp']
+        grid_coords = self.surface['grid_coords']
+        ngrids = grid_coords.shape[0]
         v_grids_e = cupy.empty([nset, ngrids])
         for i in range(nset):
-            v_grids_e[i] = 2.0*int3c2e.get_j_int3c2e_pass1(self.intopt, dms[i])
+            v_grids_e[i] = intor(self.mol, "int1e_grids", grid_coords, dm = dms[i], charge_exponents = charge_exp**2, intopt = self.intopt)
```
Comment (wxj6000) [inline gpu4pyscf/solvent/pcm.py:402]: Move the loop into `intor` and support multiple density matrices. We could possibly improve the efficiency by batching density matrices.

File: gpu4pyscf/solvent/pcm.py [hunk 2]
```diff
@@ -393,19 +394,23 @@ def _get_v(self, dms):
...
         return v_grids_e
 
     def _get_vmat(self, q):
         assert q.ndim == 2
         nset = q.shape[0]
         nao = self.mol.nao
+        charge_exp  = self.surface['charge_exp']
+        grid_coords = self.surface['grid_coords']
         vmat = cupy.empty([nset, nao, nao])
         for i in range(nset):
```
Comment (wxj6000) [inline gpu4pyscf/solvent/pcm.py:412]: Same here.

**Comment by sunqm** [comment]
* `int3c1e` here actually corresponds to the `int1e_grids` in `pyscf.Mole.intor()` . The relevant functions and files should be renamed to reflect this connection or proper docstrings should be added in int3c1e.py and other files to explain this.

* Please add a docstring at the head of int3c1e_ip.py to explain what type of integrals that ip1 and ip2 compute.

**Comment by henryw7** [comment]
> * `int3c1e` here actually corresponds to the `int1e_grids` in `pyscf.Mole.intor()` . The relevant functions and files should be renamed to reflect this connection or proper docstrings should be added in int3c1e.py and other files to explain this.
> * Please add a docstring at the head of int3c1e_ip.py to explain what type of integrals that ip1 and ip2 compute.

I added explanation of int3c1e, ip1 and ip2 to the corresponding public function. We decided to change the function name from `intor()` to ones different from pyscf function names (now called `int1e_grids()`, `int1e_grids_ip1()` and `int1e_grids_ip2()`, and save the work of consistent interface design later.

**Review by sunqm — APPROVED** [review]

**Pull request merged**